### PR TITLE
🚨🚨 Source Jira: add UserGroupsDetailed schema, fix Incremental normalization, add Incremental IssueComments, IssueWorklogs

### LINKED ISSUE #8473: Source Jira: unnesting of issue.fields in normalization fails
## Environment
- **Airbyte version**:  0.32.11-alpha
- **OS Version / Instance**: macOs
- **Deployment**: Docker
- **Source Connector and version**: Jira 0.2.15
- **Destination Connector and version**: Postgres 0.3.16
- **Severity**: Very Low / Low / **Medium** / High / Critical
- **Step where error happened**: Deploy / **Sync job** / Setup new connection / Update connector / Upgrade Airbyte

## Current Behavior
When trying to sync `board_issues` stream with Incremental + Dedup the sync failed during normalization. The reason is because the cursor field is configured to use a nested field `cursor_field = ["fields", "updated"]`

## Expected Behavior
The sync must work with incremental.

## Logs
Complete logs: [logs-1-0 (4).txt](https://github.com/airbytehq/airbyte/files/7644926/logs-1-0.4.txt)
```
oidCall):82 -   File "/usr/local/lib/python3.8/site-packages/normalization/transform_catalog/transform.py", line 35, in run
normalization - 2021-12-02 19:29:49 ERROR () LineGobbler(voidCall):82 -     self.process_catalog()
normalization - 2021-12-02 19:29:49 ERROR () LineGobbler(voidCall):82 -   File "/usr/local/lib/python3.8/site-packages/normalization/transform_catalog/transform.py", line 62, in process_catalog
normalization - 2021-12-02 19:29:49 ERROR () LineGobbler(voidCall):82 -     processor.process(catalog_file=catalog_file, json_column_name=json_col, default_schema=schema)
normalization - 2021-12-02 19:29:49 ERROR () LineGobbler(voidCall):82 -   File "/usr/local/lib/python3.8/site-packages/normalization/transform_catalog/catalog_processor.py", line 74, in process
normalization - 2021-12-02 19:29:49 ERROR () LineGobbler(voidCall):82 -     nested_processors = stream_processor.process()
normalization - 2021-12-02 19:29:49 ERROR () LineGobbler(voidCall):82 -   File "/usr/local/lib/python3.8/site-packages/normalization/transform_catalog/stream_processor.py", line 282, in process
normalization - 2021-12-02 19:29:49 ERROR () LineGobbler(voidCall):82 -     self.generate_scd_type_2_model(from_table, column_names),
normalization - 2021-12-02 19:29:49 ERROR () LineGobbler(voidCall):82 -   File "/usr/local/lib/python3.8/site-packages/normalization/transform_catalog/stream_processor.py", line 810, in generate_scd_type_2_model
normalization - 2021-12-02 19:29:49 ERROR () LineGobbler(voidCall):82 -     cursor_field=self.get_cursor_field(column_names),
normalization - 2021-12-02 19:29:49 ERROR () LineGobbler(voidCall):82 -   File "/usr/local/lib/python3.8/site-packages/normalization/transform_catalog/stream_processor.py", line 833, in get_cursor_field
normalization - 2021-12-02 19:29:49 ERROR () LineGobbler(voidCall):82 -     raise ValueError(f"Unsupported nested cursor field {'.'.join(self.cursor_field)} for stream {self.stream_name}")
normalization - 2021-12-02 19:29:49 ERROR () LineGobbler(voidCall):82 - ValueError: Unsupported nested cursor field fields.updated for stream board_issues
```

## Steps to Reproduce
1. create source jira with integration account creds
2. create a postgres
3. run sync using the specific stream with incremental

## Are you willing to submit a PR?

Remove this with your answer.

### LINKED ISSUE #14339: source-jira: pagination failure, Users stream only returns 50 records
## Environment
- **Source Connector and version**: source-jira 0.2.20

## Current Behavior
The `Users` stream only returns 50 records. This is probably related to `next_page_token` implementation as [the default number](https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-users/#api-rest-api-3-users-search-get) of returned values on this endpoint is 50. 

## Expected Behavior
The Users stream should return all available users

## Logs
[Related forum discussion](https://discuss.airbyte.io/t/cant-get-all-jira-users/1664)

### PR DESCRIPTION
**Pull request by Barsoomx**
## What
Pagination fix for Users Stream was incorrect. maxResults is capped at 1000, so if you have over 1000 users - it doesn't work. This PR fixes https://github.com/airbytehq/airbyte/issues/14339

Incremental normalization for Incremental Streams did not work due to nested fields. The field was unnested via transform() and now works correctly. This PR fixes https://github.com/airbytehq/airbyte/issues/8473

IssueComments and IssueWorklogs now can be synced incrementally, but it can be done because they are dependent on Issue stream, which can be synced incrementally and the `updated` value changes on comment or added worklog.

Users stream did not sync any user-group relationship, so 

## How
Unnesting of issue.fields

Correct pagination for Users stream. This also reverts https://github.com/airbytehq/airbyte/pull/16944 and https://github.com/airbytehq/airbyte/pull/18505 since the maxRecords field is removed.

IssueComments and IssueWorklogs now pass stream_state to Issues upstream

## 🚨 User Impact 🚨
Affected Incremental connections will have to be recreated

## Pre-merge Checklist

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-config/init/src/main/resources/seed/source_definitions.yaml [hunk 1]
```diff
@@ -606,7 +606,7 @@
 - name: Jira
   sourceDefinitionId: 68e63de2-bb83-4c7e-93fa-a8a9051e3993
   dockerRepository: airbyte/source-jira
-  dockerImageTag: 0.2.22
+  dockerImageTag: 0.2.23
```
Comment (koconder) [inline airbyte-config/init/src/main/resources/seed/source_definitions.yaml:609]: @Barsoomx do not edit this file, we will do this change when we publish. Please revert.

File: airbyte-config/init/src/main/resources/seed/source_specs.yaml [hunk 2]
```diff
@@ -5582,7 +5582,7 @@
     supportsNormalization: false
     supportsDBT: false
     supported_destination_sync_modes: []
-- dockerImage: "airbyte/source-jira:0.2.22"
+- dockerImage: "airbyte/source-jira:0.2.23"
```
Comment (koconder) [inline airbyte-config/init/src/main/resources/seed/source_specs.yaml:5585]: Please revert this change

File: airbyte-integrations/connectors/source-jira/Dockerfile [hunk 3]
```diff
@@ -12,5 +12,5 @@ RUN pip install .
 ENV AIRBYTE_ENTRYPOINT "python /airbyte/integration_code/main.py"
 ENTRYPOINT ["python", "/airbyte/integration_code/main.py"]
 
-LABEL io.airbyte.version=0.2.22
+LABEL io.airbyte.version=0.2.23
```
Comment (koconder) [inline airbyte-integrations/connectors/source-jira/Dockerfile:15]: @Barsoomx Use 0.2.3 since its a large % of changes

**Review by vincentkoc — CHANGES_REQUESTED** [review]
@Barsoomx i will run some integration tests, but please see the changes for now

File: airbyte-integrations/connectors/source-jira/Dockerfile (hunk 3 above)
Comment (Barsoomx) [reply airbyte-integrations/connectors/source-jira/Dockerfile:15]: 0.2.3 already exists. Maybe 0.3.0?

**Comment by koconder** [comment]
@Barsoomx see acceptance test failures for review

**Comment by koconder** [comment]
@Barsoomx see acceptance test failures for review

**Review by vincentkoc — CHANGES_REQUESTED** [review]
acceptance tests are failing, see comments.

**Comment by Barsoomx** [comment]
Yes, I see, working on it, thanks

File: airbyte-integrations/connectors/source-jira/Dockerfile (hunk 3 above)
Comment (koconder) [reply airbyte-integrations/connectors/source-jira/Dockerfile]: @Barsoomx sure 0.3.0 will work

**Comment by koconder** [comment]
@Barsoomx request review again once ready for me to check

**Comment by Barsoomx** [comment]
@koconder what should I do with failing `test_backward_compatibility` test? should I disable it for this version or just ignore it?
```
      backward_compatibility_tests_config:
        disable_for_version: "0.3.0"
```
for discovery test?

**Comment by koconder** [comment]
@Barsoomx you need to modify `acceptance-test-config.yml` to :
```
  spec:
    - spec_path: "source_jira/spec.json"
      backward_compatibility_tests_config:
        disable_for_version: "0.2.23"
```
I don’t have access to make changes to your PR for some reason.

**Comment by Barsoomx** [comment]
It was disabled for version "0.2.23", I bumped it for version "0.3.0", which is this PR's version right now

**Comment by Barsoomx** [comment]
Ah, I see, I have to disable previous version, not the current one. One moment

**Comment by Barsoomx** [comment]
Should I also add
```
  discovery:
    - config_path: "secrets/config.json"
      backward_compatibility_tests_config:
        disable_for_version: "0.2.23"
```

Since `TestDiscovery::test_backward_compatibility` failed due to changes in transform in several streams?

**Comment by Barsoomx** [comment]
@koconder

**Review by vincentkoc — APPROVED** [review]

**Comment by koconder** [comment]
@Barsoomx you need to give me access to your repo so i can manually upgrade the spec for the image as we are bumping the version, this will then be ready for merge.

**Comment by Barsoomx** [comment]
@koconder for some reason it doesn't allow me to allow edits to maintainers. probably because the fork was made with an org account. Do you want me to run the step 4 myself or do is it better that I re-create the fork with my account?

**Comment by koconder** [comment]
@Barsoomx please.
`airbyte-config/init/src/main/resources/seed/source_definitions.yaml`
image
`airbyte-config/init/src/main/resources/seed/source_specs.yaml`
image

**Comment by Barsoomx** [comment]
@koconder done :)

**Comment by Barsoomx** [comment]
failed because dockerhub tag already exists. should i bump it to `0.3.1` or some other workaround?

**Comment by Barsoomx** [comment]
@koconder
